src/main/java/com/ioet/bpm/people/boundaries/PersonController.java

```diff
@@ -51,17 +61,29 @@ public PersonController(PersonRepository personRepository) {
 
     @PutMapping("/{id}")
     public ResponseEntity<Person> updatePerson(@PathVariable(value = "id") String personId,
-                                             @Valid @RequestBody Person personToUpdate) {
+                                               @Valid @RequestBody Person personToUpdate) throws UnsupportedEncodingException, NoSuchAlgorithmException {
```

Comment:
Same here with exceptions, the problem is if we throw exceptions in the controller, it will show technical information to end users which is a big lack of security.